packages/flutter/test/widgets/gesture_detector_test.dart

```diff
@@ -125,222 +125,267 @@ void main() {
     expect(didEndPan, isTrue);
   });
 
-  testWidgets('Translucent', (WidgetTester tester) async {
-    bool didReceivePointerDown;
-    bool didTap;
-
-    Future<void> pumpWidgetTree(HitTestBehavior behavior) {
-      return tester.pumpWidget(
-        Directionality(
-          textDirection: TextDirection.ltr,
-          child: Stack(
-            children: <Widget>[
-              Listener(
-                onPointerDown: (_) {
-                  didReceivePointerDown = true;
-                },
-                child: Container(
+  group('Tap', () {
+    int button = 0x0;
+    void setButton(int value) {
+      button = value;
+    }
+
+    final ListVariant<int> buttonVariant = ListVariant<int>(
```

Comment:
I think it is a little too abstract for unit tests. Can we move to a simpler structure like a `for` loop over a structured data such as:
```
class TapVariant {
  final String buttonName;
  final int button;
}

Reply:
The reason to use a variant test is to avoid the possibility of leakage between loops in a structure like that.  It runs completely different tests for each variant instead of relying on the test to make sure that all the intermediate state got reset between loops. Also, each test needs to have extra `for` loops around it, which makes them more complicated and harder to read.

I could make a specific `TestVariant` subclass just for buttons if you like.

Reply:
Yeah, it seems to me that a local subclass to `TestVariant` is sufficient. It seems to me that `ListVariant` only redirects the methods to be overridden to the callbacks to be assigned.